actions/runner
# [1742] Ensure multiple composite annoations are correctly written.

This implementation uses a collector pattern to allow embedded `ExecutionContext's` to stash `Issue` objects for later processing by a non-embedded ancestor `ExecutionContext`.

### Context
Proposed fix for https://github.com/actions/runner/issues/1742

### Also:
 - Provide explicit constructor implementations for `ExecutionContext`
 - Leverage explicit constructors to solidify immutability of several `ExecutionContext` class members.
 - Use a consistent timestamp for `FinishTime` in `ExecutionContext::Complete`
 - Fixed erroneous call to `ExecutionContext::Complete` in `CompositeActionHandler.cs`
 - Addressed serialization inconsistencies in `TimelineRecord`

### Demonstration of Problem and Fix

![image](https://user-images.githubusercontent.com/81404201/225870304-0c125339-8d89-485f-b9ea-cb3fb7db14c2.png)

source: https://github.com/bbq-beets/jww3-2022a/actions/runs/4445968310 

#### Workflow files

```
name: Issue 1742 Test
on:
  workflow_dispatch:
jobs:   
  demonstrate-bug:
    runs-on: ubuntu-latest
    steps:
      - name: Demonstrate the bug on a GitHub-hosted runner.
        run:  echo "Prior to the fix being released, this should demonstrate the bug."

      - name: Composite action
        uses: konradpabjan/composite-action-test@main

  verify-fix:
    runs-on: self-hosted
    steps:
      - name: Verification of the fix on a Custom Runner Build
        run:  echo "Now, we'll try and verify the fix."

      - name: Composite action
        uses: konradpabjan/composite-action-test@main
```
source:  https://github.com/bbq-beets/jww3-2022a/blob/main/.github/workflows/verify-fix-1742.yml

```
name: 'Hello World'
description: Greeting
runs:
  using: "composite"
  steps:
    - run: echo Hello there once!
      shell: bash
    - name: Testing the Error Workflow Command
      run: echo "::error::Testing error"
      shell: bash
    - run: echo Hello there
      shell: bash
    - name: Testing the Error Workflow Command
      run: echo "::error::Testing error"
      shell: bash
    - name: Testing the Error Workflow Command
      run: echo "::warning::Testing warning"
      shell: bash
    - name: Testing the Error Workflow Command
      run: echo "::notice::Testing notice"
      shell: bash
```
source:  https://github.com/konradpabjan/composite-action-test/blob/main/action.yml
